### ISSUE TITLE
Non-existent build folder used with doubly inheriting configure preset

### ISSUE THREAD

**Issue by mpayrits** [microsoft/vscode-cmake-tools#4023]
### Brief Issue Summary

With a `CMakeUserPresets.json` and no `CMakePresets.json` where the active preset inherits from two hidden presets, the CMake vscode tools build step executes with a made-up, non-existent build folder, and consequently fails.  
The specific setup that fails is included in the Additional Information section.
Adding a `CMakePresets.json` file, even if its `configurePresets` field is empty, makes the problem go away, as does moving the specification of `binaryDir` to the `default` preset.

### CMake Tools Diagnostics

```shell
{
  "os": "win32",
  "vscodeVersion": "1.92.2",
  "cmtVersion": "1.19.49",
  "configurations": [
    {
      "folder": "c:\\repos\\cmake_repro",
      "cmakeVersion": "3.30.0",
      "configured": true,
      "generator": "Ninja",
      "usesPresets": true,
      "compilers": {
        "CXX": "C:/Program Files (x86)/Microsoft Visual Studio/2022/BuildTools/VC/Tools/MSVC/14.40.33807/bin/Hostx64/x64/cl.exe"
      }
    }
  ],
  "cpptoolsIntegration": {
    "isReady": true,
    "hasCodeModel": false,
    "activeBuildType": "",
    "buildTypesSeen": [
      "Debug"
    ],
    "requests": [],
    "responses": [],
    "partialMatches": [],
    "targetCount": 0,
    "executablesCount": 0,
    "librariesCount": 0,
    "targets": []
  },
  "settings": [
    {
      "communicationMode": "automatic",
      "useCMakePresets": "auto",
      "configureOnOpen": true
    }
  ]
}
```

### Debug Log

```shell
[main] Building folder: C:/repos/cmake_repro/build/Release 
[main] Saving open files before configure/build
[build] Starting build
[driver] Start build 
[proc] Executing command: "C:\Program Files\CMake\bin\cmake.EXE" --build C:/repos/cmake_repro/out/build/default-Release --parallel 22 --
[proc]   with environment:  # Deleted by OP
[build] Error: C:/repos/cmake_repro/out/build/default-Release is not a directory
[proc] The command: "C:\Program Files\CMake\bin\cmake.EXE" --build C:/repos/cmake_repro/out/build/default-Release --parallel 22 -- exited with code: 1
[driver] Build completed: 00:00:00.101
[cmakefileapi-parser] Read reply folder: C:\repos\cmake_repro\build\Release\.cmake\api\v1\reply
[cmakefileapi-parser] Found index files: ["cache-v2-6a79af79a33f529f12b0.json","cmakeFiles-v1-e2d2853485946773b423.json","codemodel-v2-ab181b5ee363a1d3bd47.json","directory-.-Debug-d0094a50bb2071803777.json","index-2024-08-29T14-46-54-0792.json","toolchains-v1-d7ac54a936b2bd07920c.json"]
[build] Build finished with exit code 1
[cache] Reading CMake cache file C:/repos/cmake_repro/build/Release/CMakeCache.txt
[cache] Parsing CMake cache string
[extension] [9669] cmake.build finished (returned 1)
[cache] Reading CMake cache file C:/repos/cmake_repro/build/Release/CMakeCache.txt
[cache] Parsing CMake cache string
[cpptools] The build configurations generated do not contain the active build configuration. Using "Debug" for CMAKE_BUILD_TYPE instead of "null" to ensure that IntelliSense configurations can be found
```

### Additional Information

The reproducing project consists of these two files.
[CMakeUserPresets.json](https://github.com/user-attachments/files/16801031/CMakeUserPresets.json)
[CMakeLists.txt](https://github.com/user-attachments/files/16801032/CMakeLists.txt)
Adding the following trivial file makes the problem go away.
[CMakePresets.json](https://github.com/user-attachments/files/16801050/CMakePresets.json)

**Comment by gcampbell-msft** [comment]
@mpayrits We are investigating and I have diagnosed the issue. I hope to have a PR up and merge by tomorrow so that we can get a pre-release version out with a fix for you to use. Thanks for your understanding.

**Comment by gcampbell-msft** [comment]
Could you please test the changes I've made in #4028 by installing this vsix (you'll have to modify the extension from .zip to .vsix) and confirm whether it fixes the issue? 

Thanks!
[cmake-tools.zip](https://github.com/user-attachments/files/16809854/cmake-tools.zip)

**Comment by mpayrits** [comment]
Hi, I've tried out the custom vsix and can confirm it fixes the issue 👍

**Comment by gcampbell-msft** [comment]
The fix for this issue will be released in the next hour or so in v1.20.2.

**Comment by v-frankwang** [comment]
@gcampbell-msft We verified this issue on CMake tools:v1.20.2 (pre-release) and it has been fixed.

[CMake Tools Diagnostics.txt](https://github.com/user-attachments/files/16831778/CMake.Tools.Diagnostics.txt)

Actual result：
![image](https://github.com/user-attachments/assets/9f06d1c8-e9d7-4fde-ae13-535a5278b3f1)
